codeforamerica/vita-min
# Fyst 126 handle non matching state code in url and intake

## Link to pivotal/JIRA issue
- https://codeforamerica.atlassian.net/browse/FYST-126
## Is PM acceptance required?
- Yes - don't merge until JIRA issue is accepted!
## What was done?
- The routes (URLs) for all state file pages had a state code (az/ny) in them, even though there should always be an intake in the session for the current state. Therefore, we chose to remove the state code from the routes for all pages except the landing pages and the faq pages.
- We also now delete _all_ intakes from the session when creating a new one at the beginning of the flow, not just ones for the selected state.
## How to test?
- I made sure all existing tests passed once the state code param was removed
- I added a bunch of new controller tests to make sure that the state file questions views load successfully after the change
- I imported the faqs locally and checked that those pages still load manually
- I partially went through the flows manually to make sure that they still work
- Risk Assessment
  - Almost any link in the app could be broken by this change - I am 95% sure that they aren't but not 100% so we should click on everything, especially in the header and footer.
  - We should also confirm that pages still lay out the same and that the state colors still show up correctly everywhere.
